### PR TITLE
adding precommit check

### PR DESCRIPTION
**Pull request by Borda**
## Why are these changes needed?

This effectively replace the fist check in list bellow as it runs the precommit and PR check

## Checks

- [x] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've made sure [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
- [ ] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by sonichi — COMMENTED** [review]
[notebook/automl_time_series_forecast.ipynb](https://github.com/microsoft/FLAML/pull/930/files#diff-e3988a541b7c37bda783b1e4a1a0cb8742d7f665a7135190bab664bc06e6ba5c)
is shown in the "changed files", but I can't see what's changed.

**Comment by Borda** [comment]
> is shown in the "changed files", but I can't see what's changed.

it was probably edited by the check, so trying to revert it...

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+jobs:
+
+  pre-commit-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        # for private repo - first is the checkout step, which needs to use unlimited fetch depth for pushing
+        #with:
+        #  fetch-depth: 0
```
Comment (ZviBaratz) [inline .github/workflows/pre-commit.yml:21]:
```suggestion
```

File: .github/workflows/pre-commit.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      - uses: actions/setup-python@v4
+
```
Comment (ZviBaratz) [inline .github/workflows/pre-commit.yml:23]:
```suggestion
```

File: .github/workflows/pre-commit.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      - name: set PY
```
Comment (ZviBaratz) [inline .github/workflows/pre-commit.yml:24]:
```suggestion
      - name: Set $PY environment variable
```

File: .github/workflows/pre-commit.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        run: echo "PY=$(python -VV | sha256sum | cut -d' ' -f1)" >> $GITHUB_ENV
+      - uses: actions/cache@v3
+        with:
+          path: ~/.cache/pre-commit
+          key: pre-commit|${{ env.PY }}|${{ hashFiles('.pre-commit-config.yaml') }}
+
```
Comment (ZviBaratz) [inline .github/workflows/pre-commit.yml:30]:
```suggestion
```

File: .github/workflows/pre-commit.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      - uses: pre-commit/action@v3.0.0
+        # this action also provides an additional behaviour when used in private repositories
+        # when configured with a GitHub token, the action will push back fixes to the pull request branch
+        #with:
+        #  token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (ZviBaratz) [inline .github/workflows/pre-commit.yml:35]:
```suggestion
```

**Review by ZviBaratz — APPROVED** [review]
Looks good, removed some unnecessary whitespace and comments.

**Review by ZviBaratz — APPROVED** [review]

**Review by skzhang1 — APPROVED** [review]

**Pull request merged**